src/ripple/shamap/impl/SHAMapInnerNode.cpp

```diff
@@ -239,51 +946,72 @@ SHAMapInnerNode::shareChild(
     int m,
     std::shared_ptr<SHAMapAbstractNode> const& child)
 {
-    assert((m >= 0) && (m < 16));
+    assert((m >= 0) && (m < branchFactor));
     assert(cowid_ != 0);
     assert(child);
     assert(child.get() != this);
 
-    mChildren[m] = child;
+    assert(!isEmptyBranch(m));
+    hashesAndChildren_.getChildren()[*getChildIndex(m)] = child;
 }
 
 SHAMapAbstractNode*
 SHAMapInnerNode::getChildPointer(int branch)
 {
-    assert(branch >= 0 && branch < 16);
+    assert(branch >= 0 && branch < branchFactor);
+    assert(!isEmptyBranch(branch));
 
     std::lock_guard lock(childLock);
-    return mChildren[branch].get();
+    return hashesAndChildren_.getChildren()[*getChildIndex(branch)].get();
 }
 
 std::shared_ptr<SHAMapAbstractNode>
 SHAMapInnerNode::getChild(int branch)
 {
-    assert(branch >= 0 && branch < 16);
+    assert(branch >= 0 && branch < branchFactor);
+    assert(!isEmptyBranch(branch));
 
     std::lock_guard lock(childLock);
-    return mChildren[branch];
+    return hashesAndChildren_.getChildren()[*getChildIndex(branch)];
+}
+
+SHAMapHash const&
+SHAMapInnerNode::getChildHash(int m) const
+{
+    assert((m >= 0) && (m < branchFactor));
+    if (auto const i = getChildIndex(m))
+        return hashesAndChildren_.getHashes()[*i];
+
+    static SHAMapHash const zero = [] {
+        SHAMapHash zero;
+        zero.zero();
+        return zero;
+    }();
+    return zero;
```

Comment:
Per my earlier comment, I think this needs to be a `static const` member of `SHAMapInnerNode`.

Reply:
Fixed in `2fa4a8d59e [fold] Doxygen comments and misc cleanups 1:`